geps/gep-1767/index.md

```diff
@@ -0,0 +1,701 @@
+    // 
+    // Support: Extended
+    //
+    // +optional
+    CORS *HTTPCORSFilter `json:"cors,omitempty"`
+}
+
+// HTTPCORSFilter defines a filter that responds to the
+// cross-origin request based on HTTP header.
+type HTTPCORSFilter struct {
+    // AllowOrigins indicates whether the response can be shared with 
+    // requested resource from the given `Origin`.
+    // 
+    // A wildcard indicates that the requests from all `Origin` are allowed.
+    //
+    // When responding to a credentialed requests, the gateway must specify 
+    // an origin in the value of the Access-Control-Allow-Origin response header, 
+    // instead of specifying the * wildcard.
+    //
+    // Input:
+    //   Origin: https://foo.example
+    //
+    // Config:
+    //   allowOrigins: ["*"]
+    //
+    // Output:
+    //   Access-Control-Allow-Origin: *
+    //
+    // For a simple cross-origin interaction, the gateway sets value of the header 
+    // Access-Control-Allow-Origin same as the `Origin` header provided by the client.
+    //
+    // The status code of a successful response to a "preflight" request is an OK status (i.e., 204 or 200). 
+    // For the "preflight" request, if the request `Origin` does not match the configured allowed origins,
+    // the gateway will return a response with error status (e.g., 403). 
+    // Moreover, the gateway will omit the relevant cross-origin response headers.
+    // Then, the client will not send actual cross-origin request.
+    // 
+    // Input:
+    //   Origin: https://foo.example
+    //
+    // Config:
+    //   allowOrigins: ["https://foo.example", "http://foo.example", "https://test.example", "http://test.example"]
+    //
+    // Output:
+    //   Access-Control-Allow-Origin: https://foo.example
+    //
+    // Support: Extended
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=64
+    AllowOrigins []string `json:"allowOrigins,omitempty"`
+ 
+    // AllowCredentials indicates whether the actual cross-origin request 
+    // allows to include credentials.
+    //
+    // The only valid value for the header `Access-Control-Allow-Credentials` is true (case-sensitive).
+    //
+    // The default value of the config AllowCredentials is false.
+    // If the credentials are not allowed in cross-origin requests, 
+    // the gateway will omit the header `Access-Control-Allow-Credentials` entirely.
+    //
+    // Input:
+    //   Origin: https://foo.example
+    //
+    // Config:
+    //   allowCredentials: true
+    //
+    // Output:
+    //   Access-Control-Allow-Origin: https://foo.example
+    //   Access-Control-Allow-Credentials: true
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:default=false
+    AllowCredentials bool `json:"allowCredentials,omitempty"`
+
+    // AllowMethods indicates which HTTP methods are supported 
+    // for accessing the requested resource.
+    // The method is case-sensitive.
```

Comment:
Again, we need to be specific about the exact values. In this case, we can say something like:

```suggestion
    // Valid values are any method defined by the HTTP RFC (add number and link here).
    // Method names are case sensitive per the HTTP RFCs, so these values are also case-sensitive.
```

Reply:
(See https://www.rfc-editor.org/rfc/rfc2616#section-5.1.1)